azimuth_schedule_operator/operator.py

```diff
@@ -46,15 +51,68 @@ async def startup(settings, **kwargs):
 
 
 @kopf.on.cleanup()
-async def cleanup(**kwargs):
+async def cleanup(**_):
     if K8S_CLIENT:
         await K8S_CLIENT.aclose()
     LOG.info("Cleanup complete.")
 
 
-@kopf.on.create(registry.API_GROUP, "schedule")
-@kopf.on.update(registry.API_GROUP, "schedule")
-@kopf.on.resume(registry.API_GROUP, "schedule")
-async def schedule_changed(body, name, namespace, labels, **kwargs):
+async def get_reference(namespace: str, ref: schedule_crd.ScheduleRef):
+    resource = await K8S_CLIENT.api(ref.api_version).resource(ref.kind)
+    object = await resource.fetch(ref.name, namespace=namespace)
+    return object
+
+
+async def delete_reference(namespace: str, ref: schedule_crd.ScheduleRef):
+    resource = await K8S_CLIENT.api(ref.api_version).resource(ref.kind)
+    await resource.delete(ref.name, namespace=namespace)
+
+
+async def check_for_delete(namespace, schedule: schedule_crd.Schedule):
+    now = datetime.datetime.now(datetime.timezone.utc)
+    if now >= schedule.spec.not_after:
+        LOG.info(f"Attempting delete for {namespace} and {schedule.metadata.name}.")
+        await delete_reference(namespace, schedule.spec.ref)
+        await update_schedule(
+            schedule.metadata.name, namespace, ref_delete_triggered=True
+        )
+    else:
+        LOG.info(f"No delete for {namespace} and {schedule.metadata.name}.")
+
+
+async def update_schedule(
+    name: str,
+    namespace: str,
+    ref_exists: bool = None,
+    ref_delete_triggered: bool = None,
+):
+    now = datetime.datetime.now(datetime.timezone.utc)
+    now_string = now.strftime("%Y-%m-%dT%H:%M:%SZ")
+    status_updates = dict(updated_at=now_string)
```

Comment:
`updatedAt`?

Reply:
Oops, I missed that when I fixed up the other two. I have the silent errors on those, added some bits in the functional test.